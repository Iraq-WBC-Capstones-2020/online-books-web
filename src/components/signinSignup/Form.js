import React from 'react';
import { MDBCol, MDBCard, MDBCardBody, MDBInput, MDBBtn } from 'mdbreact';

export default function Form({ formData, onSubmit }) {
  const { title, button, extraAnchor, inputFields } = formData;

  return (
    <MDBCol md="5" className="mt-5 ml-5">
      <MDBCard className="mt-5 log z-depth-0">
        <MDBCardBody className="mx-4">
          <div className="text-center bg-gray-400">
            <h3 className="mb-5">
              <strong>{title}</strong>
            </h3>
          </div>
          <form onSubmit={onSubmit}>
            {inputFields.map((input) => (
              <MDBInput
                key={input.id}
                label={input.label}
                group
                type={input.type}
                name={input.name}
                validate
                error="wrong"
                success="right"
              />
            ))}
            {extraAnchor && (
              <p className="d-flex justify-content-end pb-3">
                <a href={extraAnchor.href} className="ml-1">
                  {extraAnchor.title}
                </a>
              </p>
            )}
            <div className="text-center mb-3">
              <MDBBtn
                type="submit"
                color="blue"
                className="rounded-pill text-white z-depth-1a w-48"
              >
                {button.title}
              </MDBBtn>
            </div>
          </form>
          <div className="row my-3 d-flex justify-content-center"></div>
        </MDBCardBody>
      </MDBCard>
    </MDBCol>
  );
}
