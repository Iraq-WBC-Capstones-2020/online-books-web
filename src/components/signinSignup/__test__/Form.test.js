import React from 'react';
import renderer from 'react-test-renderer';
import Form from './../Form';

test('Form snapshot', () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
