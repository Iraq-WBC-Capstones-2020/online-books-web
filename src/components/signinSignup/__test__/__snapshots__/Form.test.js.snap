// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot 1`] = `
Array [
  <div
    className="container"
    data-test="container"
  >
  <div data-test="col" className="mt-5 ml-5 col-md-5">
      <div className="mt-5 log z-depth-0"  data-test="card">
        <div className="mx-4 card-body"  data-test="card">
          <div className="text-center ">
            <h3 className="mb-5">
              <strong>{title}</strong>
            </h3>
          </div>
          <form onSubmit={onSubmit}>
            {inputFields.map((input) => (
              <input
                key={input.id}
                label={input.label}
                group
                type={input.type}
                name={input.name}
                validate
                error="wrong"
                success="right"
              />
            ))}
            {extraAnchor && (
              <p className="d-flex justify-content-end pb-3">
                <a href={extraAnchor.href} className="ml-1">
                  {extraAnchor.title}
                </a>
              </p>
            )}
            <div className="text-center mb-3">
              <button
                type="submit"
                color="blue"
                className="rounded-pill text-white z-depth-1a w-48"
              >
                {button.title}
              </button>
            </div>
          </form>
          <div className="row my-3 d-flex justify-content-center"></div>
        </div>
      </div>
    </div>
   
]
`;
